models/recipe.model.js

```diff
@@ -0,0 +1,81 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const { User, UserSchema } = require("./user.model");
+
+const TagSchema = new Schema({
+    id: {
+        type: Number,
+        required: true,
+    },
+    name: {
+        type: String,
+        required: true,
+    },
+});
+
+module.exports = TagSchema;
+module.exports = Tag = mongoose.model("tag", TagSchema);
+
+const IngredientSchema = new Schema({
+    id: {
+        type: Number,
+        required: true,
+    },
+    name: {
+        type: String,
+        required: true,
+    },
+    amount: {
+        type: Number,
+        required: true,
+    },
+    unit: {
+        type: String,
+        required: true,
+    },
+});
+module.exports = IngredientSchema;
+module.exports = Ingredient = mongoose.model("ingredient", IngredientSchema);
+
+const RecipeSchema = new Schema({
+    id: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: false,
+    },
+    ownerID: {
+        type: String,
+        required: false,
+    },
+    title: {
+        type: String,
+        required: true,
+    },
+    ingredients: {
+        type: [IngredientSchema], //ingredient or ingredient schema?
+        required: true,
+    },
+    steps: {
+        type: [String],
+        required: true,
+    },
+    notes: {
```

Comment:
notes should probably be a User field because each user would have their own specific notes for a recipe

Reply:
OH i see so more like a comment? I was thinking they'd be a recipe-wide note from the original uploader (be careful on the salt, etc) and then we could include another comments mechanism